### PR TITLE
Add test framework

### PR DESCRIPTION
**Pull request by joekiller**
The point of this PR is to add a useful set of tests which can be expanded upon as new features arrive. The test framework is [Jest](https://jestjs.io/en/).

Current goals:

  - Enable asserting the result of SteamUser.chatMessage() calls

Features thus far:

  - Executing in CI
  - Current tests are in `src/classes/__tests__/` which is common Jest convention. It makes imports easy for what is being tested generally because you import ../classname.

### CODE REVIEW

**Comment by idinium96** [comment]
Please only create a new branch from the `development` branch and create a pull request to the `development` branch.

Thank you.
https://github.com/idinium96/tf2autobot/blob/master/.github/contributing.md

**Comment by joekiller** [comment]
Damn I thought I rebased on development.

**Comment by joekiller** [comment]
@idinium96 I think the PR just defaulted to master. I swapped it.

File: src/classes/__tests__/MyHandler.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import MyHandler from '../MyHandler';
+import Bot from '../Bot';
+import BotManager from '../BotManager';
+import SteamID from 'steamid';
+import SteamUser from 'steam-user';
+
+jest.mock('SteamUser');
+
+test('onMessage', () => {
+    SteamUser.chatMessage.mockResolvedValue();
```
Comment (idinium96) [inline src/classes/__tests__/MyHandler.ts:10]: maybe you should add `const client: SteamUser = new SteamUser()` before this line and replace `SteamUser.chatMessage` with `client.chatMessage` then let see if this check failure disappears.

**Comment by idinium96** [comment]
I think I will not add this.

**Comment by joekiller** [comment]
Hrm I didn't intend this to be ready. To keep track of progress would you prefer I make an issue that the branch will fulfill? At this point the code is supposed to be an example of where the starting point of testing is.

**Comment by joekiller** [comment]
Yes rtfm. https://github.com/idinium96/tf2autobot/blob/master/.github/contributing.md#pull-requests

I'll make an issue called add testing that'll track my progress and then open the pr

**Pull request closed without merging**